src/components/UnlockButton.tsx

```diff
@@ -1,12 +1,26 @@
 import React from 'react'
-import { Button, useWalletModal } from '@pancakeswap-libs/uikit'
-import { useWallet } from '@binance-chain/bsc-use-wallet'
+import { Button, ConnectorId, useWalletModal } from '@pancakeswap-libs/uikit'
+import { useWeb3React } from '@web3-react/core'
+import { WalletConnectConnector } from '@web3-react/walletconnect-connector'
+
+import { injected, walletconnect } from 'connectors'
 import useI18n from 'hooks/useI18n'
 
 const UnlockButton = (props) => {
   const TranslateString = useI18n()
-  const { connect, reset } = useWallet()
-  const { onPresentConnectModal } = useWalletModal(connect, reset)
+  const { account, activate, deactivate, connector } = useWeb3React()
+  const handleLogin = (connectorId: ConnectorId) => {
+    if (connectorId === 'walletconnect') {
+      if (connector instanceof WalletConnectConnector && connector.walletConnectProvider?.wc?.uri) {
```

Comment:
What this if block is for ?

Reply:
It's for walletconnect provider init issue

Reply:
When users sign in again using walletconnect, it produces bug